### ISSUE TITLE
Question: timed animation

### ISSUE THREAD

**Issue by alexreardon** [chenglou/react-motion#140]
Is there anyway to have a timed animation? Stiffness and damping would be auto calculated based on distance / time.

**Comment by chenglou** [comment]
Not currently, since it's against the goal of interruptible animation. Might expose these in the future, I'll see. I'd rather people adjust the spring to achieve a good "feel" rather than putting a random number. Here's something that might help you find the parameters you want: https://cdn.rawgit.com/chenglou/react-motion/e8f42dcd9678a8cea8648a3cf4f994583a99e7f7/demos/demo5/index.html

**Comment by alexreardon** [comment]
My current goal is to build a time lapse. I need to animate each step in full before going to the next step. Currently I am guessing based on spring and dampness; but the time for animation could be different as distances between start and finish positions can vary.

**Comment by chenglou** [comment]
What kind of time lapse specifically? I'm actually planning to support sequential animation, so maybe checking the completion each time isn't even the best way to do it for you.

**Comment by alexreardon** [comment]
Yep; time might not be the best way. Really I want to know when the previous animation finished: https://github.com/chenglou/react-motion/issues/139

**Comment by chenglou** [comment]
Sequence it is then?

So there are two types of sequences here. One is pre-prop, where your prop goes to the next destination value when the current one finishes. The other is per-"set of props", where you want all the props (each of which might have a different end time) to finish all before starting another set. Which one do you need?

@threepointone

**Comment by alexreardon** [comment]
Currently I have a control that sets the current state. I want to know when the animation finishes so that it can set the next state (which would be animated to)

**Comment by alexreardon** [comment]
So I won't have a series of animations set up ready to go. Rather, I just need to know when the current animation is finished so I can update the application state. Similar outcome to chained animations though

**Comment by AgentME** [comment]
I'm trying to figure out how to speed up animations for tests, and I'm not sure how to adjust the stiffness and damping to speed up the animation. If I have {stiffness: 300, damping: 50}, is there a simple formula I can do to make the same animation but 2x faster? I've tried arbitrarily increasing stiffness and/or damping, but I always end up with no perceptible difference, or making the animation overshoot its end horribly (and taking longer).

**Comment by alexreardon** [comment]
@AgentME you can do this:

We use a requestAnimation polyfill for testing [raf]((https://github.com/chrisdickinson/raf)
Raf uses set `timeout` in place of `window.requestAnimationFrame`

(Alternatively you could just replace `window.requestAnimationFrame` with `setTimeout`)

When testing you can use [sinon](https://github.com/sinonjs/sinon) to hyjack the window timers (see `sinon.useFakeTimers`). This will allow you to 'flush' the animation queue.

Example:

```js
// simple example - a component that renders out a div and moves it. 
describe('DraggableTranslate', () => {
    let clock;

    beforeEach(() => {
        clock = sinon.useFakeTimers();
    });

    afterEach(() => {
        clock.restore();
    });
    
    it('should translate by the offset', () => {
        const offset = {
            x: 100,
            y: 200
        };
        const component = TestUtils.renderIntoDocument(
            <YourComponent  offset={offset}/>
        );

       // flush the animation queue
        clock.tick(1000);

        const [el] = TestUtils.scryRenderedDOMComponentsWithTag(component, 'div');
        expect(el.style.transform).to.eql(`translate(${offset.x}px, ${offset.y}px)`);
   });
});
```

**Comment by AgentME** [comment]
Thanks, though well also I'm still interested in knowing how to speed up an animation for an application. I have no idea how exactly I should change the numbers if I want to have the same animation style but faster.

**Comment by chenglou** [comment]
Closing in favor of #265 (will release a new helper that allows you set the "duration" in a more intuitive way, even without the help of demo 5).
